### ISSUE TITLE
Cannot pygmentize JSON with Unicode escapes (regression in Pygments==2.7.0)

### ISSUE THREAD

**Issue by vytas7** [pygments/pygments#1544]
```
$ echo -e '{\n     "French": "Fran\\u00e7ais",\n     "bug": true\n}\n' 
{
     "French": "Fran\u00e7ais",
     "bug": true
}
```

Using `Pygments==2.7.0`:
```
$ pygmentize -V
Pygments version 2.7.0, (c) 2006-2020 by Georg Brandl.
$ echo -e '{\n     "French": "Fran\\u00e7ais",\n     "bug": true\n}\n' | pygmentize -l json -F raiseonerror 

*** Error while highlighting:
pygments.filters.ErrorToken: "
   (file "<... my virtualenv ...>/lib/python3.8/site-packages/pygments/filters/__init__.py", line 783, in filter)
*** If this is a bug you want to report, please rerun with -v.
{
```
Removing `-F raiseonerror` does produce highlighted output, but it looks malformed.

Using `Pygments==2.6.1` works as expected (output colours looked OK in my terminal):
```
$ pygmentize -V
Pygments version 2.6.1, (c) 2006-2019 by Georg Brandl.
$ echo -e '{\n     "French": "Fran\\u00e7ais",\n     "bug": true\n}\n' | pygmentize -l json -F raiseonerror 
{
     "French": "Fran\u00e7ais",
     "bug": true
}
```

**Comment by birkenfeld** [comment]
Thanks for the report, I've released 2.7.1 with the fix in place.

**Comment by vytas7** [comment]
That was quick. Thanks!
